enterprise/frontend/src/embedding-sdk/components/public/InteractiveDashboard/ConnectedDashboard.tsx

```diff
@@ -127,19 +130,41 @@ const ConnectedDashboardInner = ({
 
 export const ConnectedDashboard = connector<
   ComponentType<ConnectedDashboardProps & ReduxProps>
->(({ dashboardId: initId, ...rest }) => {
-  const { id: dashboardId, isLoading } = useValidatedEntityId({
+>(({ dashboardId: initialDashboardId, ...rest }) => {
+  const { id: resolvedDashboardId, isLoading } = useValidatedEntityId({
     type: "dashboard",
-    id: initId,
+    id: initialDashboardId,
   });
 
+  const errorPage = useSdkSelector(getErrorPage);
+  const dispatch = useSdkDispatch();
+  useEffect(() => {
+    if (resolvedDashboardId) {
+      dispatch(setErrorPage(null));
```

Comment:
since we use `errorPage` from above we need to clear it if client change the ID to the resolved ones.